client/src/components/Filter.jsx

```diff
@@ -34,19 +34,10 @@ const Filter = ({ filter, setFilter, saveFilterToSessionStorage }) => {
             name="hold_color"
             value={filter.hold_color}
             onChange={event => {
-              setFilter({
-                hold_color: event.target.value,
-                level: '',
-                sector: '',
-                climb_result: '',
-                projectedOnly: false,
-              });
+              setFilter({ ...filter, hold_color: event.target.value });
               saveFilterToSessionStorage({
+                ...filter,
                 hold_color: event.target.value,
```

Comment:
could this be simplified to `saveFilterToSessionStorage(filter)`?

Reply:
Hi Paul, your suggestion didn't work as then the sessionfilter value would always be one selection "behind". However I did a little code cleanup and merged "set filter" and "saveFilterToSessionStorage" into one function that I call now when the user changes the filter selection.

Reply:
surprising but how could I have known :)  Your clean-up looks good. The only thing I'm wondering is whether the session storage actually needs to be updated every time or if there is like an event triggered when leaving the find page that could be suitable for that.

Reply:
I will put this on the nice2have list. :-)

Reply:
Sounds good. Meanwhile I read a bit and it doesn't seem unusual to update session storage all the time (I couldn't find an example like yours, though). One thing I came across was updating session storage via useEffect hook on the state, which l liked. see here: [https://www.darrenlester.com/blog/syncing-react-state-and-session-storage](url)